src/router.rs

```diff
@@ -131,15 +136,27 @@ impl Router {
     /// Route will match any method, including gibberish.
     /// In case of ambiguity, handlers specific to methods will be preferred.
     pub fn any<S: AsRef<str>, H: Handler, I: AsRef<str>>(&mut self, glob: S, handler: H, route_id: I) -> &mut Router {
-        self.mut_inner().wildcard.add(glob.as_ref(), Box::new(handler));
+        self.mut_inner().wildcard.add(glob.as_ref(), Arc::new(handler));
         self.route_id(route_id.as_ref(), glob.as_ref());
         self
     }
 
     fn recognize(&self, method: &method::Method, path: &str)
-                     -> Option<Match<&Box<Handler>>> {
-        self.inner.routers.get(method).and_then(|router| router.recognize(path).ok())
-            .or(self.inner.wildcard.recognize(path).ok())
+                      -> Result<Match<&Arc<Handler>>, IronError> {
```

Comment:
In `handle` we would wrap it in `IronError` anyway (resulting in the same outside API), this is about the `recognize` method.

You can do `pub use Error::NoRoute` to make the `NoRoute` variant importable at the same location.